cmake/developer_package/python_requirements.cmake

```diff
@@ -97,7 +97,10 @@ function(ov_check_pip_packages)
 
     if(PYTHONINTERP_FOUND)
         execute_process(
-            COMMAND ${PYTHON_EXECUTABLE} -c "import pkg_resources ; pkg_resources.require(open('${ARG_REQUIREMENTS_FILE}', mode='r'))"
+            COMMAND ${PYTHON_EXECUTABLE} -c "import pkg_resources ;
+            requirements_raw = open('${ARG_REQUIREMENTS_FILE}', mode='r').readlines() ;
+            requirements = [package for package in requirements_raw if not package.startswith('-c')] ;
+            pkg_resources.require(requirements)"
```

Comment:
I suppose we need concrete versions here

Reply:
I just pushed a commit with concrete requirement for just `onnx=1.12`, it's far from perfect but I want to see if it fixes the issues.

Reply:
The current parser now works correctly - just like before, but while taking `constraints.txt` into account.